packages/line/src/LineSlices.js

```diff
@@ -11,30 +11,62 @@ import PropTypes from 'prop-types'
 import pure from 'recompose/pure'
 import LineSlicesItem from './LineSlicesItem'
 
-const LineSlices = ({
-    slices,
-    height,
-    showTooltip,
-    hideTooltip,
-    theme,
-    tooltip,
-    tooltipFormat,
-}) => (
-    <g>
-        {slices.map(slice => (
-            <LineSlicesItem
-                key={slice.id}
-                slice={slice}
-                height={height}
-                showTooltip={showTooltip}
-                hideTooltip={hideTooltip}
-                theme={theme}
-                tooltipFormat={tooltipFormat}
-                tooltip={tooltip}
-            />
-        ))}
-    </g>
-)
+class LineSlices extends React.Component {
+    render() {
+        const {
+            slices,
+            height,
+            showTooltip,
+            hideTooltip,
+            theme,
+            tooltip,
+            tooltipFormat,
+        } = this.props
+        // sliceWidth: half the distance from the previous point + half the distance to the next point
+        let sliceWidth
+        // xOffset: half the distance from the previous point
+        let xOffset
+        let lineSlicesItems = []
+        for (let i = 0; i < slices.length; i++) {
```

Comment:
This might be intense for load and calculation for many points? Would it be better to instead say at larger slices it is defaulted? Or a tiered process?

Reply:
I think it would also depend on the chart width in the window as well. A smaller div would mean you have less space for the points to work?